core/src/mindustry/logic/LAssembler.java

```diff
@@ -61,8 +61,14 @@ public static String write(Seq<LStatement> statements){
         String[] lines = data.split("\n");
         int index = 0;
         for(String line : lines){
-            //comments
-            if(line.startsWith("#") || line.isEmpty()) continue;
+            // comments
+            int commentIdx = line.indexOf('#');
+            if (commentIdx != -1) {
```

Comment:
```suggestion
            if (commentIdx != -1){
```

Reply:
Yeah, that's the "formatting aside" remark on the linked issue.